clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -242,6 +242,9 @@ public synchronized boolean ensurePartitionAdded(TopicPartition tp) {
             // We should enter this branch in an error state because if this partition is already in the transaction,
             // there is a chance that the corresponding batch is in retry. So we must let it completely flush.
             if (!(partitionsInTransaction.contains(tp) || isPartitionPending(tp))) {
+                log.debug("partitionsInTransaction: [{}], newPartitionsInTransaction: [{}], " +
```

Comment:
Is this intentional or were you just using this for your own debugging? If it's intentional, we should probably add the log prefix and make the message a little clearer.